### PR TITLE
Allow to prepare response outside of Request

### PR DESCRIPTION
**Pull request by Alex-Bond**
Hi!

This change required to receive body data for PSR-7 response passing (not echo).

| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | none

### CODE REVIEW

File: src/Response.php [hunk 1]
```diff
@@ -1031,7 +1031,7 @@ public function getIsEmpty()
      * The default implementation will convert [[data]] into [[content]] and set headers accordingly.
      * @throws InvalidConfigException if the formatter for the specified format is invalid or [[format]] is not supported
      */
-    protected function prepare()
+    public function prepare()
```
Comment (samdark) [inline src/Response.php]: The API with this change doesn't look good. How about calling it in a getter instead?

File: src/Response.php [hunk 2]
```diff
@@ -328,19 +328,29 @@ public function setContent($content)
 
     /**
      * Sends the response to the client.
+     * @param bool $return Send to stdout or return object. If set to `true` method will return prepared Response object.
+     * @return Response
+     * @throws HeadersAlreadySentException
+     * @throws InvalidConfigException
      */
-    public function send()
+    public function send($return = false)
```
Comment (rob006) [inline src/Response.php]: Now we have `send()` that does not send. :D

Comment (samdark) [reply src/Response.php]: How about `getData()`?

Comment (Alex-Bond) [reply src/Response.php]: It's a little confusing. Technically you send it but not in stdout but to the variable.

Comment (Alex-Bond) [reply src/Response.php]: Like with print_r function.

Comment (Alex-Bond) [reply src/Response.php]: Also logically it suppose to call events that you already have in send() when i requesting filled response...

Comment (samdark) [reply src/Response.php]: Aha. Then maybe `process()` that returns data? Then `send()` will look like `echo process()`.

Comment (Alex-Bond) [reply src/Response.php]: @samdark that can work. What about events `EVENT_BEFORE_SEND` and `EVENT_AFTER_SEND`? Technically we sending content somewhere. Or these events apply only to stdout print?

Comment (samdark) [reply src/Response.php]: Sending events should be called for sending only. We can add two additional events for processing.  Roadrunner-like server and traditional server are two similar but different models so there may be need to do things a bit differently.

**Comment by samdark** [comment]
#11

**Comment by samdark** [comment]
Goind to implement proper PSR usage.

**Pull request closed without merging**